tests/queries/0_stateless/03357_system_errors_query_id.sh

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+# Tags: no-parallel
+
+CUR_DIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+# shellcheck source=../shell_config.sh
+. "$CUR_DIR"/../shell_config.sh
+
+query_id="01545_system_errors-$CLICKHOUSE_DATABASE"
+$CLICKHOUSE_CLIENT --query_id="$query_id" -q "select query_id" &>/dev/null
+
+retry=0
+max_retries=30
+while [[ $($CLICKHOUSE_CLIENT -q "SELECT count(*) FROM system.errors WHERE query_id = '$query_id' and code = 47") -ne 1 ]]; do
```

Comment:
This test is still racy, you need to retry the first query in the loop as well, otherwise if between `SELECT query_id` and `SELECT FROM system.errors WHERE query_id = ?` will be another query that will throw this error, this loop will never finishes, and actually this loop does not make sense without retrying the query that cause the `47` error.